### ISSUE TITLE
[SPIR-V] Use proper bitwidth in getValueOne and getValueZero

### ISSUE THREAD

**Issue by gongminmin** [Microsoft/DirectXShaderCompiler#2044]
Compile this shader with -T ps_6_2 -enable-16bit-types -spirv
```
half MySaturate(half x)
{
	return saturate(x);
}

float4 PS_Main(half x : TEXCOORD0) : SV_Target
{
	return MySaturate(x);
}
```
returns this error

> fatal error: generated SPIR-V is invalid: GLSL.std.450 FClamp: expected types of all operands to be equal to Result Type
>   %16 = OpExtInst %half %1 FClamp %15 %float_0 %half_0x1p_0
> 
> note: please file a bug report on https://github.com/Microsoft/DirectXShaderCompiler/issues with source code if possible

But calling saturate(x) directly inside PS_Main doesn't have this error.

**Comment by ehsannas** [comment]
That's odd... Thanks for reporting @gongminmin

**Comment by ehsannas** [comment]
I actually get the same error even if saturate(x) is called directly inside PS_Main.

**Comment by ehsannas** [comment]
these two functions:
getValueOne
getValueZero

are currently using 32-bit int/uint/float regardless of the given QualType.
The intention of these functions has been to generate a constant value of 1 or 0 with the given QualType. They should be updated to take the bitwidth of the given QualType into account.

**Comment by jaebaek** [comment]
Fixed by #2068
